### PR TITLE
MOEN-25537: Implement fetchCards API for flutter

### PR DESCRIPTION
**Pull request by msoumya-engg-sdk**
### Jira Ticket

https://moengagetrial.atlassian.net/browse/MOEN-25537

### Description

Implement fetchCards API in flutter: https://moengagetrial.atlassian.net/wiki/spaces/MS/pages/3806986336/Sync+API+For+Cards

### CODE REVIEW

File: cards/lib/src/model/card_data.dart
Comment (arshiya-moengage) [inline cards/lib/src/model/card_data.dart:1]: Lets keep the name consistent with native, in android this class is  named `CardsInfo`, please check on the iOS side and have the same name

Comment (arshiya-moengage) [inline cards/lib/src/model/card_data.dart:1]: The existing class `CardsData` serves the same purpose, as discussed for fetching all cards the value for `category` can be set to `all`. Please remove this class

Comment (kkgowtham-engg-sdk) [reply cards/lib/src/model/card_data.dart:1]: Updated

File: cards/android/src/main/kotlin/com/moengage/flutter/cards/PlatformMethodCallHandler.kt [hunk 1]
```diff
@@ -248,4 +271,8 @@ class PlatformMethodCallHandler(
             Logger.print(LogLevel.ERROR, t) { "$tag getUnClickedCardsCount() : " }
         }
     }
+
+    private fun getCardPayload(cardsData: CardData?, payload: String): JSONObject {
```
Comment (arshiya-moengage) [inline cards/android/src/main/kotlin/com/moengage/flutter/cards/PlatformMethodCallHandler.kt:275]: the parameter name can be `cardData`

Comment (kkgowtham-engg-sdk) [reply cards/android/src/main/kotlin/com/moengage/flutter/cards/PlatformMethodCallHandler.kt]: Updated

**Review by arshiya-moengage — APPROVED** [review]

**Pull request merged**